### PR TITLE
Add pod prune for api v2

### PR DESCRIPTION
**Pull request by sujil02**
Add pod prune for api v2.

Add the ability to prune pods for api v2,
Includes the addition of force flag, for client-side prompt.

Signed-off-by: Sujil02 <sushah@redhat.com>

### CODE REVIEW

File: pkg/api/server/register_pods.go [hunk 1]
```diff
@@ -51,13 +51,14 @@ func (s *APIServer) registerPodsHandlers(r *mux.Router) error {
...
+	//    type: boolean
+	//    description : force removal of a running pods
 	// responses:
 	//   200:
-	//     description: tbd
-	//     schema:
-	//       type: object
-	//       additionalProperties:
-	//         type: string
+	//   200:
```
Comment (vrothberg) [inline pkg/api/server/register_pods.go:60]: Drop this line.

File: pkg/api/server/register_pods.go [hunk 2]
```diff
@@ -51,13 +51,14 @@ func (s *APIServer) registerPodsHandlers(r *mux.Router) error {
 	// summary: Prune unused pods
 	// produces:
 	// - application/json
+	//  - in: query
+	//    name: force
+	//    type: boolean
+	//    description : force removal of a running pods
```
Comment (vrothberg) [inline pkg/api/server/register_pods.go:57]:
```suggestion
	//    description : force removal of a running pod
```

File: pkg/bindings/pods/pods.go [hunk 3]
```diff
@@ -98,17 +98,23 @@ func Pause(ctx context.Context, nameOrID string) (*entities.PodPauseReport, erro
 	return &report, response.Process(&report)
 }
 
-// Prune removes all non-running pods in local storage.
-func Prune(ctx context.Context) error {
+// Prune by default removes all non-running pods in local storage.
+// And with force set true removes all pods.
+func Prune(ctx context.Context, force *bool) ([]*entities.PodPruneReport, error) {
```
Comment (vrothberg) [inline pkg/bindings/pods/pods.go:103]: Do we _need_ force to be a pointer? If not, let's make it call by value.

File: cmd/podmanV2/pods/prune.go [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+	registry.Commands = append(registry.Commands, registry.CliCommand{
+		Mode:    []entities.EngineMode{entities.ABIMode, entities.TunnelMode},
+		Command: pruneCommand,
+		Parent:  podCmd,
+	})
+	flags := pruneCommand.Flags()
+	flags.BoolVarP(&pruneOptions.Force, "force", "f", false, "Force removal of all running pods.  The default is false")
+}
+
+func prune(cmd *cobra.Command, args []string) error {
```
Comment (vrothberg) [inline cmd/podmanV2/pods/prune.go:39]: We need to check the arguments here as v1 does:

```
~ $ podman pod prune 234908234908234 23423423
Error: `podman pod prune` takes no arguments
```

File: pkg/api/handlers/libpod/pods.go [hunk 5]
```diff
@@ -230,13 +230,31 @@ func PodRestart(w http.ResponseWriter, r *http.Request) {
 func PodPrune(w http.ResponseWriter, r *http.Request) {
 	var (
 		runtime = r.Context().Value("runtime").(*libpod.Runtime)
+		decoder = r.Context().Value("decoder").(*schema.Decoder)
+		reports []*entities.PodPruneReport
 	)
-	pruned, err := runtime.PrunePods()
+	query := struct {
+		Force bool `schema:"force"`
```
Comment (mheon) [inline pkg/api/handlers/libpod/pods.go:237]: This does not match the usage of Force for pruning containers or volumes, we should drop this. Force is client-side-only and removes the yes/no continue prompt.

Comment (sujil02) [reply pkg/api/handlers/libpod/pods.go:237]: @baude @jwhonce I thought we agreed to add the force for prune pod. 
Should I go ahead had change this to the client-side only?

File: cmd/podmanV2/pods/prune.go (hunk 4 above)
Comment (sujil02) [reply cmd/podmanV2/pods/prune.go:40]: fixed.

File: pkg/bindings/pods/pods.go (hunk 3 above)
Comment (sujil02) [reply pkg/bindings/pods/pods.go:103]: _force_ need not be a pointer. However, I noticed we have used pointers in other references of similar use case hence went with maintaining consistency eg: https://github.com/containers/libpod/blob/master/pkg/bindings/pods/pods.go#L155.

Should I still go ahead and change it?

Comment (vrothberg) [reply pkg/bindings/pods/pods.go:103]: Pointers are usually used to find out whether a certain value has been set on the CLI. This allows for distinguishing it from being set (true/false) and not being set (nil), which allows for using defaults in backend code. As we are not doing that in this case, there's no need to have a pointer.

File: pkg/api/handlers/libpod/pods.go (hunk 5 above)
Comment (sujil02) [reply pkg/api/handlers/libpod/pods.go]: Fixed the force flag. @mheon PTAL

File: pkg/bindings/pods/pods.go (hunk 3 above)
Comment (sujil02) [reply pkg/bindings/pods/pods.go]: Thank you. Wrt today discussion force flag for pod prune is not required here hence marking resolved.

**Comment by rhatdan** [comment]
LGTM
@vrothberg @mheon @TomSweeneyRedHat PTAL

File: cmd/podmanV2/pods/prune.go [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+	var (
+		errs utils.OutputErrors
+	)
+	if len(args) > 0 {
+		return errors.Errorf("`%s` takes no arguments", cmd.CommandPath())
+	}
+	if !pruneOptions.Force {
+		reader := bufio.NewReader(os.Stdin)
+		fmt.Println("WARNING! This will remove all volumes not used by at least one container.")
```
Comment (vrothberg) [inline cmd/podmanV2/pods/prune.go:52]: "This will remove all volumes not used by at least one container"...looks like a copy&paste error.

Comment (sujil02) [reply cmd/podmanV2/pods/prune.go]: Upps. Fixed it.

**Comment by rhatdan** [comment]
@sujil02 Needs a rebase.

**Comment by sujil02** [comment]
> @sujil02 Needs a rebase.

Yeah on it. It Needs some test fixes as well with the rebase.

**Comment by sujil02** [comment]
Done. Changed the commit messages as CLI force flag edits are no longer required.

**Pull request merged**